src/state/connection-mixin.ts

```diff
@@ -78,6 +80,12 @@ export const connectionMixin = <T extends Constructor<HassBaseEl>>(
               target
             )) as Promise<ServiceCallResponse>;
           } catch (err) {
+            if (
+              err.error.code === ERR_CONNECTION_LOST &&
+              serviceCallWillDisconnect(domain, service)
+            ) {
+              throw err;
```

Comment:
Throw? Not return?

Reply:
I wasn't sure how all the existing callers would handle it if they got back nothing instead of the promise so throw + suppressing the notification in the UI seemed like the way to go.  

Better ideas?

Reply:
I think almost nothing in the frontend uses the return value of `callService`, I introduced it for the scenes editor a year ago, if they do, Typescript should be able to tell you the issues :-)

If we throw, it will not be caught by the caller, as we previously never thrown errors, but caught errors here. I would return undefined to keep current behavior.

Reply:
(and move it after the dev console log)

Reply:
done here https://github.com/home-assistant/frontend/pull/8937

Will check it in the morning.

late here. g'night

Reply:
Thanks! and Good night!